unblob/report.py

```diff
@@ -181,6 +181,7 @@ class ChunkReport(Report):
     end_offset: int
     size: int
     is_encrypted: bool
+    metadata: dict = attr.ib(factory=dict)
```

Comment:
just wondering if we want to validate metadata dict, do we want to enforce that key is a string and value is of a certain type, or we are ok we anything, even nested meta-data?

Reply:
Maybe it would be great to somehow have a "namespace" or at least some convention on metadata variable naming?

What if we want to push data from multiple headers, or permissions, etc?

Reply:
I'm ok with enforcing a convention on metadata variable naming. Having a namespace would be too complicated since we can't foresee the metadata field names used by handlers.

I would enforce that metadata is a dict without nested data, keys must be strings and values must be base types.

I would convey information about files created (timestamps, permissions, owner) with something different since it involves way more complex structures.

Reply:
Added a validator. See 99944e3